.azure/pipelines/jobs/default-build.yml

```diff
@@ -160,8 +160,8 @@ jobs:
       - script: df -h
         displayName: Disk size
     - ${{ if eq(parameters.agentOs, 'macOS') }}:
-      - script: sudo xcode-select -s /Applications/Xcode_12.2.app/Contents/Developer
-        displayName: Use XCode 12.2
+      - script: sudo xcode-select -s /Applications/Xcode_12.5.1.app/Contents/Developer
+        displayName: Use XCode 12.5.1
```

Comment:
I'm not positive but don't think we use XCode directly at all. This may exist only for the googletest submodule.

@sdmaclea why were the old lines added in your #27264 PR❔ I want to make sure we aren't breaking something w/ this move (even though everything builds and tests fine).